File: LogiOS/TrustarApp/Database/user/Embedded/BinHeaderAndStatus.swift
//
//  BinHeaderAndStatus.swift
//  TrustarApp
//
//  Created by DionSoftware on 2022/02/17.
//  Copyright © 2022 DionSoftware. All rights reserved.
//

import Foundation

class BinHeaderAndStatus {
    var header: BinHeader
    var truck: Truck
    var status: BinStatus
    
    init(header: BinHeader, truck: Truck, status: BinStatus) {
        self.header = header
        self.truck = truck
        self.status = status
    }
}

File: LogAndroid/TrustarApp/src/main/kotlin/jp/co/toukei/log/trustar/db/user/embedded/BinHeaderAndStatus.kt
package jp.co.toukei.log.trustar.db.user.embedded

import androidx.room.Embedded
import jp.co.toukei.log.lib.util.CompareItem
import jp.co.toukei.log.trustar.db.user.entity.BinHeader
import jp.co.toukei.log.trustar.db.user.entity.BinStatus
import jp.co.toukei.log.trustar.db.user.entity.Truck

class BinHeaderAndStatus(
        @Embedded @JvmField val header: BinHeader,
        @Embedded @JvmField val truck: Truck,
        @Embedded @JvmField val status: BinStatus
) : CompareItem<BinHeaderAndStatus> {

    override fun sameItem(other: BinHeaderAndStatus): Boolean {
        return other.header.sameItem(header) && other.status.sameItem(status) && other.truck.sameItem(truck)
    }
}
